assets/js/base/hooks/use-collection-data.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useMemo } from '@wordpress/element';
+import {
+	useQueryStateByContext,
+	useQueryStateByKey,
+	useCollection,
+} from '@woocommerce/base-hooks';
+import { useCollectionDataContext } from '@woocommerce/base-context/collection-data-context';
+import { useDebounce } from 'use-debounce';
+import { find, sortBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { useShallowEqual } from './use-shallow-equal';
+
+const buildCollectionDataQuery = ( collectionDataQueryState ) => {
+	const query = collectionDataQueryState;
+
+	if ( collectionDataQueryState.calculate_attribute_counts ) {
+		query.calculate_attribute_counts = sortBy(
+			collectionDataQueryState.calculate_attribute_counts.map(
+				( { taxonomy, queryType } ) => {
+					return {
+						taxonomy,
+						query_type: queryType,
+					};
+				}
+			),
+			[ 'taxonomy', 'query_type' ]
+		);
+	}
+
+	return query;
+};
+
+export const useCollectionData = ( {
+	queryAttribute,
+	queryPrices,
+	queryState,
+	context,
+} ) => {
+	const collectionDataContext = useCollectionDataContext();
+	context = context || collectionDataContext;
+
+	const [ collectionDataQueryState ] = useQueryStateByContext( context );
```

Comment:
I added this to avoid lumping the new data under `page`, since `page` is used for the products query. The new data I'm storing shouldn't be added to those queries, only the collection data queries.

Reply:
`useQueryContext` is just a string used to reference the context for data stored in the query state store for a given component.  Yes the products query is using this as it should because it's sharing the same query state store data as the collection data filters is it not?  The query state for the filters is still used by the products query to know what data to retrieve is it not?  So you still can have different keys for the collection-data query state within the `context` but I don't think you should be creating another query-state context here.  

I get you want to keep this query state separate from the one that `useCollection` is using, but the problem with this approach is it doesn't scale.  When we start supporting multiple instances of All products block we'd have to do something like this:

```jsx
<CollectionDataContextProvider value='page-collection-data-1'>
    <QueryStateContext value='product-grid-1'>
        <Block/>
    </QueryStateContext/>
<CollectionDataContextProvider/>
```
Since you can still supply the specific context to the various query state hooks here, what about doing something like this instead:

```js
let context = useQueryStateContext();
context = `${ context }-collection-data`;
```
You still get the uniqueness while also connecting it to the context shared between various components in the given query state context tree?

Reply:
Ah! Thanks, I thought it had to be registered to be used.